test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,200 @@
+(** These tests for issue #1658, which follows PR #1655 which introduces a store
+    version bump from V1 to V2 when a writer instance adds a V2-only pack entry
+    (including nodes and commits, but not contents). *)
+
+open! Import
+open Common
+
+(** {2 Preamble} *)
+
+let src = Logs.Src.create "tests.version_bump" ~doc:"Test pack version bump"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Util = struct
+  (** Following are generic utils *)
+
+  let exec_cmd = Common.exec_cmd
+  let tmp_dir () = Filename.temp_file "test_pack_version_bump_" ""
+
+  (** Make a directory; dir is assumed to not exist; parent is assumed to exist *)
+  let mkdir p =
+    assert (not (Sys.file_exists p));
+    Filename.quote_command "mkdir" [ p ] |> fun cmd ->
+    exec_cmd cmd |> function
+    | Ok () -> ()
+    | Error n ->
+        Fmt.failwith "Command `%s' exited with non-zero code %d\n" cmd n
+
+  (** Copy src to dst; dst is assumed to not exist *)
+  let copy_dir src dst =
+    assert (not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" [ "-R"; src; dst ] |> fun cmd ->
+    exec_cmd cmd |> function
+    | Ok () -> ()
+    | Error n ->
+        Fmt.failwith
+          "Failed to set up test env; command `%s' exited with non-zero code %d\n"
+          cmd n
+
+  (* NOTE we try to avoid dependence on Fpath in the following, by
+     using standard OCaml functions *)
```

Comment:
We already have a dependency on `Fpath`, so this seems unnecessary.

Reply:
Yeah, probably unnecessary. Agreed. I removed the comment.